### PR TITLE
Fix: handling of Elasticsearch aliases

### PR DESCRIPTION
**Pull request by scottinet**
# Description

Kuzzle should be able to handle index aliases using its formalism and pointing to another place.

Example/How to reproduce the bug:
* create an index named `foobar`
* alias it to `&foo.bar` 

Expected result: Kuzzle should correctly map the alias to `{ index: 'foo', collection: 'bar'}` and use it as any other non-alias indexes

Without this patch, Kuzzle incorrectly maps the alias to: `{ index: 'foo', collection: '&foo.bar'}`

# Changes

The `listAliases` method in the `elasticsearch` service now returns the following array of objects: 
```js
[ 
  {name: '<origin ES index>', index: '<inferred index name>', collection: '<inferred index collection>'},
  {name: '<origin ES index>', index: '<inferred index name>', collection: '<inferred index collection>'},
  // ...
]
```

The `name` property is not currently used, but I found it interesting to keep it for later uses.

Kuzzle's index cache bootstrap now correctly uses that `listAliases` method to get non-listed indexes by `cat.indices` and complete its cache with its results.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "kuzzle",
   "author": "The Kuzzle Team <support@kuzzle.io>",
-  "version": "2.3.0",
+  "version": "2.3.1",
```
Comment (alexandrebouthinon) [inline package.json:4]: You should maybe update your local master branch :slightly_smiling_face: 
```suggestion
  "version": "2.3.2",
```

Comment (alexandrebouthinon) [reply package.json:4]: And generate again the package-lock.json

Comment (alexandrebouthinon) [reply package.json:4]: Never mind we choose what fix we want to release first

Comment (scottinet) [reply package.json:4]: dang

Comment (scottinet) [reply package.json:4]: @alexandrebouthinon > err... are you sure?
https://github.com/kuzzleio/kuzzle/blob/master/package.json

Comment (Aschen) [reply package.json:4]: Talking about this on Slack https://kaliop.slack.com/archives/G4LN18NLQ/p1593763809492900?thread_ts=1593762619.490900&cid=G4LN18NLQ

**Pull request merged**